handlers/server_spec_handler.go

```diff
@@ -23,14 +28,44 @@ func (h *Handler) ServerVersionHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// if r.Method != http.MethodGet {
-	// 	w.WriteHeader(http.StatusNotFound)
-	// 	return
+	//      w.WriteHeader(http.StatusNotFound)
+	//      return
 	// }
 
+	// compare the server build with the target build
+	res, err := h.checkLatestVersion(version.Build)
+	if err != nil {
+		logrus.Errorln(err)
+	} else {
+		// Add "Latest" and "Outdated" fields to the response
+		version.Latest = res.Current
+		version.Outdated = &res.Outdated
+	}
+
 	w.Header().Set("Content-Type", "application/json")
 
-	err := json.NewEncoder(w).Encode(version)
+	err = json.NewEncoder(w).Encode(version)
 	if err != nil {
 		logrus.Errorf("unable to send data: %v", err)
 	}
 }
+
+// checkLatestVersion takes in the current server version compares it with the target
+// and returns the result (latest.CheckResponse)
+func (h *Handler) checkLatestVersion(serverVersion string) (*latest.CheckResponse, error) {
+	githubTag := &latest.GithubTag{
+		Owner:      mesheryGitHubOrg,
+		Repository: mesheryGitHubRepo,
+	}
+
+	// Compare current running Meshery server version to the latest available Meshery release on GitHub.
+	res, err := latest.Check(githubTag, serverVersion)
```

Comment:
Not needed, we already have a server version endpoint, please use that.

Reply:
I searched through the repository and I could only find something related to latest version checking in the mershyctl package as it was pointed out by leecalcote in the description of issue #1753 . Can you please point out the endpoint, actually I am still exploring the repository :smile: .

Reply:
@kushthedude, the task at-hand here is to take the currently running server version from the `/api/server/version` endpoint, retrieve the version number of the latest release from GitHub and compare them. This functionality was recently incorporated into `mesheryctl version`. It would be nice to let users of the UI know if they are a version behind.

Reply:
Then we can improve the version endpoint to show this, rather than having additional code in both mesheryctl and UI to do this task.